feat(email): send over ssl and append outgoing mails to sent folder

Thunderbottom opened the pull request:
* add `use_ssl_for_outgoing` to use port 465 instead of the default smtp 587 port and send mails over ssl

* remove Administrator from `email_list`

* append outgoing emails to sent folder

port of: #8383

documentation: https://github.com/frappe/erpnext_com/pull/638

Co-authored-by: Saurabh Palande <saurabh6790@gmail.com>

Thunderbottom committed:
fix: use filter for removing administrator from email list

Signed-off-by: Chinmay D. Pai <chinmaydpai@gmail.com>
Co-authored-by: Saurabh Palande <saurabh6790@gmail.com>

Thunderbottom committed:
feat(email): allow ssl for outgoing emails, append to sent folder

* dont always use default outgoing account for sending
* allow ssl for outgoing emails

Signed-off-by: Chinmay D. Pai <chinmaydpai@gmail.com>

surajshetty3416 committed:
Merge branch 'develop' into email_ssl

surajshetty3416 committed:
refactor: Commonify sent mail checks

-and fix formatting

surajshetty3416 reviewed (changes requested):
@Thunderbottom waiting for documentation.

Thunderbottom committed:
fix: move email status check to fix travis

mail status wasn't being set when the check was run, so the output for
any() would always be false, and none of the mails would have status set
to sent in the backend. moving the check to a point after setting status
for each email should fix this issue

Signed-off-by: Chinmay D. Pai <chinmaydpai@gmail.com>

scmmishra commented:
@Mergifyio backport version-12-hotfix
